cirq-core/cirq/transformers/routing/mapping_manager.py

```diff
@@ -14,154 +14,196 @@
+
+        `self.physical_qid_to_int[self.int_to_physical_qid[i]] == i` for each i.
+        """
+        return self._int_to_physical_qid
+
+    @property
+    def logical_qid_to_int(self) -> Dict['cirq.Qid', int]:
+        """Mapping of logical qubits, that were part of the initial mapping, to unique integers."""
+        return self._logical_qid_to_int
 
     @property
-    def inverse_map(self) -> Dict['cirq.Qid', 'cirq.Qid']:
-        """The mapping of physical qubits (keys) to logical qubits (values)."""
-        return self._inverse_map
+    def int_to_logical_qid(self) -> List['cirq.Qid']:
+        """Inverse mapping of unique integers to corresponding physical qubits.
+
+        `self.logical_qid_to_int[self.int_to_logical_qid[i]] == i` for each i.
+        """
+        return self._int_to_logical_qid
+
+    @property
+    def logical_to_physical(self) -> np.ndarray:
+        """The mapping of logical qubit integers to physical qubit integers.
+
+        Let `lq: cirq.Qid` be a logical qubit. Then the corresponding physical qubit that it
+        maps to can be obtained by:
+        `self.int_to_physical_qid[self.logical_to_physical[self.logical_qid_to_int[lq]]]`
+        """
+        return self._logical_to_physical
+
+    @property
+    def physical_to_logical(self) -> np.ndarray:
+        """The mapping of physical qubits integers to logical qubits integers.
+
+        Let `pq: cirq.Qid` be a physical qubit. Then the corresponding logical qubit that it
+        maps to can be obtained by:
+        `self.int_to_logical_qid[self.physical_to_logical[self.physical_qid_to_int[pq]]]`
+        """
+        return self._physical_to_logical
 
     @property
-    def induced_subgraph(self) -> nx.Graph:
-        """The induced subgraph on the set of physical qubits which are part of `self.map`."""
-        return self._induced_subgraph
+    def induced_subgraph_int(self) -> nx.Graph:
+        """Induced subgraph on physical qubit integers present in `self.logical_to_physical`."""
+        return self._induced_subgraph_int
 
-    def dist_on_device(self, lq1: 'cirq.Qid', lq2: 'cirq.Qid') -> int:
+    def dist_on_device(self, lq1: int, lq2: int) -> int:
         """Finds distance between logical qubits 'lq1' and 'lq2' on the device.
 
         Args:
-            lq1: the first logical qubit.
-            lq2: the second logical qubit.
+            lq1: integer corresponding to the first logical qubit.
+            lq2: integer corresponding to the second logical qubit.
 
         Returns:
             The shortest path distance.
         """
-        return self._distances[self._map[lq1]][self._map[lq2]]
+        return self._distances[self.logical_to_physical[lq1]][self.logical_to_physical[lq2]]
 
-    def can_execute(self, op: 'cirq.Operation') -> bool:
-        """Finds whether the given operation acts on qubits that are adjacent on the device.
+    def can_execute(self, lq1: int, lq2: int) -> bool:
```

Comment:
The name of this function no longer matches its arguments. I would expect `is_adjacent` or similar.

Reply:
Updated the name to `is_adjacent`